spec/aruba/api/filesystem_spec.rb

```diff
@@ -0,0 +1,746 @@
+            it { expect(destination).to be_an_existing_directory }
+          end
+
+          context 'when source is nested directory' do
+            let(:source) { 'src.d/subdir.d' }
+            let(:destination) { 'dst.d/' }
+
+            it { expect(destination).to be_an_existing_directory }
+          end
+        end
+      end
+
+      context 'when destination is existing' do
+        context 'when source is list of files' do
+          before(:each) { create_test_files(source) }
+
+          context 'when destination is directory' do
+            let(:source) { %w(file1.txt file2.txt file3.txt) }
+            let(:destination) { 'file.d' }
+            let(:destination_files) { source.map { |s| File.join(destination, s) } }
+
+            before :each do
+              Aruba.platform.mkdir(File.join(@aruba.aruba.current_directory, destination))
+            end
+
+            before :each do
+              @aruba.copy source, destination
+            end
+
+            it { source.each { |s| expect(destination_files).to all be_an_existing_file } }
+          end
+
+          context 'when destination is not a directory' do
+            let(:source) { %w(file1.txt file2.txt file3.txt) }
+            let(:destination) { 'file.txt' }
+
+            before(:each) { create_test_files(destination) }
+
+            it { expect { @aruba.copy source, destination }.to raise_error ArgumentError, "Multiples sources can only be copied to a directory" }
+          end
+
+          context 'when a source is the same like destination' do
+            let(:source) { 'file1.txt' }
+            let(:destination) { 'file1.txt' }
+
+            before(:each) { create_test_files(source) }
+
+            # rubocop:disable Metrics/LineLength
+            it { expect { @aruba.copy source, destination }.to raise_error ArgumentError, %(same file: #{File.expand_path(File.join(@aruba.aruba.current_directory, source))} and #{File.expand_path(File.join(@aruba.aruba.current_directory, destination))}) }
+            # rubocop:enable Metrics/LineLength
+          end
+
+          context 'when a fixture is destination' do
+            let(:source) { '%/copy/file.txt' }
+            let(:destination) { '%/copy/file.txt' }
+
+            it { expect { @aruba.copy source, destination }.to raise_error ArgumentError, "Using a fixture as destination (#{destination}) is not supported" }
+          end
+        end
+      end
+
+      context 'when source is non-existing' do
+        it { expect { @aruba.copy source, destination }.to raise_error ArgumentError }
+      end
+    end
+  end
+
+  context '#write_file' do
+    it 'writes file' do
+      @aruba.write_file(@file_name, '')
+
+      expect(File.exist?(@file_path)).to eq true
+    end
+  end
+
+  context '#write_fixed_size_file' do
+    it "should write a fixed sized file" do
+      @aruba.write_fixed_size_file(@file_name, @file_size)
+      expect(File.exist?(@file_path)).to eq true
```

Comment:
this style conflicts with line235. Both are valid, but should probably only use one style?

Reply:
That would be the end goal, but it'll be a long process :slightly_smiling_face: